app/controllers/application_controller.rb

```diff
@@ -1,30 +1,15 @@
 class ApplicationController < ActionController::Base
 
   before_action :authenticate_user!
-
-  helper_method :current_user,
-                :logged_in?
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
   private
 
-  def authenticate_user!
-    unless current_user
-      cookies[:redirect_path] = request.path if request.get?
-      redirect_to login_path, alert: 'Verify your Email and Password, please'
-    end
-
-    cookies[:email] = current_user&.email
-  end
-
-  def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
-  end
-
-  def logged_in?
-    current_user.present?
+  def after_sign_in_path_for(user)
+    current_user.is_a?(Admin) ? admin_tests_path : (stored_location_for(user) || root_path)
```

Comment:
```suggestion
    user.admin? ? admin_tests_path : (stored_location_for(user) || root_path)
```
стоит использовать входяший аргумент (`user`) и механизм проверки вынести в модель